odl/discr/grid.py

```diff
@@ -1025,23 +980,31 @@ def __array__(self, dtype=None):
         return self.points().astype(dtype)
 
     def __repr__(self):
-        """Return ``repr(self)``."""
+        """Return ``repr(self)``.
+
+        Examples
+        --------
+        >>> g = odl.RectGrid([-1, 0, 3], [2, 4, 5], [5], [2, 4, 7])
+        >>> g
+        RectGrid([-1.,  0.,  3.], [ 2.,  4.,  5.], [ 5.], [ 2.,  4.,  7.])
+        """
         if self.is_uniform:
             ctor = 'uniform_grid'
             posargs = [self.min_pt, self.max_pt, self.shape]
             posmod = [array_str, array_str, '']
-            with npy_printoptions(precision=4):
-                inner_str = signature_string(posargs, [], mod=[posmod, ''])
-            return '{}({})'.format(ctor, inner_str)
         else:
             ctor = self.__class__.__name__
             posargs = self.coord_vectors
             posmod = array_str
-            inner_str = signature_string(posargs, [], sep=[',\n', ', ', ', '],
-                                         mod=[posmod, ''])
-            return '{}(\n{}\n)'.format(ctor, indent(inner_str))
 
-    __str__ = __repr__
+        with npy_printoptions(precision=REPR_PRECISION):
```

Comment:
Hm, right. A better alternative might be to use a function `repr_precision()` that caches Numpy's default value for precision and returns the current value if it has changed, otherwise `REPR_PRECISION`. That would at least implement the left hand side of your inequality chain.